service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,182 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.entity.GiftCertificate;
+import com.epam.esm.dao.entity.Orders;
+import com.epam.esm.dao.entity.User;
+import com.epam.esm.dao.repository.GiftCertificateRepository;
+import com.epam.esm.dao.repository.OrdersRepository;
+import com.epam.esm.dao.repository.TagRepository;
+import com.epam.esm.dao.repository.UserRepository;
+import com.epam.esm.dao.specification.gift.GetGiftCertificatesByIdSpecification;
+import com.epam.esm.dao.specification.tag.GetMostWidelyUsedTagSpecification;
+import com.epam.esm.dao.specification.user.GetAllUsersSpecification;
+import com.epam.esm.dao.specification.user.GetUserByIdSpecification;
+import com.epam.esm.service.UserService;
+import com.epam.esm.service.builder.certificate.GiftCertificateDtoBuilder;
+import com.epam.esm.service.builder.order.OrdersBuilder;
+import com.epam.esm.service.builder.order.OrdersDtoBuilder;
+import com.epam.esm.service.builder.tag.TagDtoBuilder;
+import com.epam.esm.service.builder.user.UserDtoBuilder;
+import com.epam.esm.service.dto.OrdersDto;
+import com.epam.esm.service.dto.PageDto;
+import com.epam.esm.service.dto.TagDto;
+import com.epam.esm.service.dto.UserDto;
+import com.epam.esm.service.exception.EmptyOrderException;
+import com.epam.esm.service.exception.EntityNotFoundException;
+import com.epam.esm.service.exception.InsufficientFundInAccount;
+import com.epam.esm.service.exception.PageNotValidException;
+import com.epam.esm.service.exception.ServiceException;
+import com.epam.esm.service.validator.PageValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class UserServiceImpl implements UserService {
+  private final UserRepository userRepository;
+  private final OrdersRepository ordersRepository;
+  private final GiftCertificateRepository giftCertificateRepository;
+  private final UserDtoBuilder userDtoBuilder;
+  private final OrdersBuilder ordersBuilder;
+  private final GiftCertificateDtoBuilder giftCertificateDtoBuilder;
+  private final OrdersDtoBuilder ordersDtoBuilder;
+  private final TagDtoBuilder tagDtoBuilder;
+  private final TagRepository tagRepository;
+  private final PageValidator pageValidator;
+  private static final int FIST_ELEMENT = 0;
+
+  @Autowired
+  public UserServiceImpl(
```

Comment:
monster constructor))

Reply:
You are absolutely right. This construction is messy.  (fixed)